index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
```

Comment:
it does:
```
If <file>s are given, it reformats the files. If -i is specified
together with <file>s, the files are edited in-place. Otherwise, the
result is written to the standard output.

USAGE: clang-format [options] [<file> ...]
```

so can we just run the binary a single time? any particular reason you want to do it this way?